src/ORM/FieldType/DBForeignKey.php

```diff
@@ -36,9 +34,16 @@ class DBForeignKey extends DBInt
      *
      * @config
      * @var int
+     * @deprecated 5.2.0 Use dropdown_lazyload_threshold instead
      */
     private static $dropdown_field_threshold = 100;
 
+    /**
+     * Number of related objects to show in a dropdown before it switches to using lazyloading
+     * This will also be used as the lazy load limit
+     */
+    private static int $dropdown_lazyload_threshold = 10;
```

Comment:
Why not just lazy-load by default? And why is the threshold so low? And why would the lazy load limit also be tied to this? It feels like I'm misunderstanding something here.

Reply:
UX is worse when lazy loading because there's a noticeable loading delay while the AJAX request is being made.

Yes 10 is on the low side, though you only (from memory) 12 options before you get a scroll bar which ideally we should avoid.

> And why would the lazy load limit also be tied to this

So that it acts as a "max number of options to show in the dropdown" config - perhaps we should rename this config to something like "dropdown_max_options_displayed"?

Reply:
I think the lazy load limit should not be tied to this.

The threshold is there for performance reasons - so it should be quite high (50 or 100) as loading that number of records should be fine performance-wise in most scenarios.

The default lazy-load limit in the field should be used in terms of determining how many records to display when lazy-loading, since that limit is tied to the records displayed when the user searches.

Reply:
@emteknetnz This has still not been addressed

Reply:
As discussed in person, the default lazy-load limit will be set to 100. The config _should_ pass through by setting `setLazyLoadLimit()` as is being done now, but the default for the config should also be 100.

Reply:
Not done yet?

Reply:
Updated